### PR TITLE
Add testcases for Kafka receiver/exporter and confmap providers

### PR DESCRIPTION
**Pull request by rapphil**
**Description:** Update terraform branch of the test framework with the content from the dev branch.

* Add support to test confmap providers.
* Add support to test kafka receiver and exporter.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by bryan-aguilar — APPROVED** [review]

**Review by vasireddy99 — APPROVED** [review]

File: terraform/testcases/configmap_provider_https/otconfig.tpl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        match_type: regexp
+        action: update
+        operations:
+          - action: add_label
+            new_label: "testingId"
+            new_value: "${testing_id}"
+
+exporters:
+  logging:
+    loglevel: debug
```
Comment (PaurushGarg) [inline terraform/testcases/configmap_provider_https/otconfig.tpl:24]: should we use loglevel:debug here (and in other configs) or as below:
```
exporters:
  logging:
    verbosity:
```

File: docs/config-map-providers.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+               | SampleApp +------->    Collector  +-------------> Cloudwatch Metrics |
+               |           |       |               |             |                    |
+               +----^------+       +---------------+             +----------+---------+
+                    ^                                                       |
+                    |                                                       |
+                    |                                                       |
+                    |                                                       |
+                    |                +-----------------+                    |
+                    |                |                 |                    |
+                    +----------------+     Validator   <--------------------+
```
Comment (PaurushGarg) [inline docs/config-map-providers.md:49]: does validator also communicates to sample app in this test?

Comment (rapphil) [reply docs/config-map-providers.md:49]: yes, how otherwise it would exercise the endpoints?

Comment (rapphil) [reply docs/config-map-providers.md:49]: Example: https://github.com/aws-observability/aws-otel-test-framework/blob/terraform/validator/src/main/resources/validations/spark-otel-metric-validation.yml#L3

```

-
  validationType: "cw-metric"
  httpPath: "/outgoing-http-call"
  httpMethod: "get"
  callingType: "http"
  expectedMetricTemplate: "ENHANCED_EXPECTED_METRIC"

```
In the example above, validator will call `/outgoin-http-call` to create metrics.

File: terraform/testcases/configmap_provider_https/otconfig.tpl (hunk 1 above)
Comment (rapphil) [reply terraform/testcases/configmap_provider_https/otconfig.tpl:24]: I will do this in a different pr because I want to fix all occurrences.

**Pull request merged**